### ISSUE TITLE
Server 0.7.0.0 - Consistent Server Crashing.

### ISSUE THREAD

**Issue by ChrisMayhew** [StrangeLoopGames/EcoIssues#3882]
[ServerCrash.txt](https://github.com/StrangeLoopGames/EcoIssues/files/1692641/ServerCrash.txt)

It seems to be roughly every half an hour that the server is online that it'll crash with the error log that has been attached above. This may be a Mono issue but I just thought I'd report the error log so someone could have a look over it.

**Comment by MarjieAVolk** [comment]
```
--BEGIN DUMP--
Dump Time
02/04/2018 09:38:15

Exception
System.AggregateException: One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object
  at Eco.World.PersistentChunk.Tick () [0x00057] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at Eco.World.PersistentChunk.TryTick () [0x00008] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at Eco.World.WorldPlugin+<>c__DisplayClass24_0.<Run>b__0 (Eco.World.PersistentChunk chunk) [0x0000c] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at System.Threading.Tasks.Parallel+<>c__DisplayClass30_0`2[TSource,TLocal].<ForEachWorker>b__0 (System.Int32 i) [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Parallel+<>c__DisplayClass17_0`1[TLocal].<ForWorker>b__1 () [0x0020b] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.InnerInvoke () [0x0000f] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.InnerInvokeWithArg (System.Threading.Tasks.Task childTask) [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task+<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0 (System.Object <p0>) [0x00086] in <997dde78663c4b4bbf4a1f2b8d296905>:0
   --- End of inner exception stack trace ---
  at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00011] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.Wait (System.Int32 millisecondsTimeout, System.Threading.CancellationToken cancellationToken) [0x00043] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.Wait () [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Parallel.ForWorker[TLocal] (System.Int32 fromInclusive, System.Int32 toExclusive, System.Threading.Tasks.ParallelOptions parallelOptions, System.Action`1[T] body, System.Action`2[T1,T2] bodyWithState, System.Func`4[T1,T2,T3,TResult$
  at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal] (TSource[] array, System.Threading.Tasks.ParallelOptions parallelOptions, System.Action`1[T] body, System.Action`2[T1,T2] bodyWithState, System.Action`3[T1,T2,T3] bodyWithStateAndIndex, System$
  at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal] (System.Collections.Generic.IEnumerable`1[T] source, System.Threading.Tasks.ParallelOptions parallelOptions, System.Action`1[T] body, System.Action`2[T1,T2] bodyWithState, System.Action`3[T1,T$
  at System.Threading.Tasks.Parallel.ForEach[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Action`1[T] body) [0x0001c] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at Eco.World.WorldPlugin.Run () [0x00026] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at Eco.Server.PluginManager+<>c__DisplayClass18_0.<StartPlugins>b__0 (System.Object x) [0x00000] in <170e11321e414586b2de019f749a752b>:0
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00025] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.ThreadHelper.ThreadStart (System.Object obj) [0x0000f] in <997dde78663c4b4bbf4a1f2b8d296905>:0
---> (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object
  at Eco.World.PersistentChunk.Tick () [0x00057] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at Eco.World.PersistentChunk.TryTick () [0x00008] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at Eco.World.WorldPlugin+<>c__DisplayClass24_0.<Run>b__0 (Eco.World.PersistentChunk chunk) [0x0000c] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at System.Threading.Tasks.Parallel+<>c__DisplayClass30_0`2[TSource,TLocal].<ForEachWorker>b__0 (System.Int32 i) [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Parallel+<>c__DisplayClass17_0`1[TLocal].<ForWorker>b__1 () [0x0020b] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.InnerInvoke () [0x0000f] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.InnerInvokeWithArg (System.Threading.Tasks.Task childTask) [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task+<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0 (System.Object <p0>) [0x00086] in <997dde78663c4b4bbf4a1f2b8d296905>:0 <---



Inner Exception
System.AggregateException: One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object
  at Eco.World.PersistentChunk.Tick () [0x00057] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at Eco.World.PersistentChunk.TryTick () [0x00008] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at Eco.World.WorldPlugin+<>c__DisplayClass24_0.<Run>b__0 (Eco.World.PersistentChunk chunk) [0x0000c] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at System.Threading.Tasks.Parallel+<>c__DisplayClass30_0`2[TSource,TLocal].<ForEachWorker>b__0 (System.Int32 i) [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Parallel+<>c__DisplayClass17_0`1[TLocal].<ForWorker>b__1 () [0x0020b] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.InnerInvoke () [0x0000f] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.InnerInvokeWithArg (System.Threading.Tasks.Task childTask) [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task+<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0 (System.Object <p0>) [0x00086] in <997dde78663c4b4bbf4a1f2b8d296905>:0
   --- End of inner exception stack trace ---
  at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00011] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.Wait (System.Int32 millisecondsTimeout, System.Threading.CancellationToken cancellationToken) [0x00043] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.Wait () [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Parallel.ForWorker[TLocal] (System.Int32 fromInclusive, System.Int32 toExclusive, System.Threading.Tasks.ParallelOptions parallelOptions, System.Action`1[T] body, System.Action`2[T1,T2] bodyWithState, System.Func`4[T1,T2,T3,TResult$
  at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal] (TSource[] array, System.Threading.Tasks.ParallelOptions parallelOptions, System.Action`1[T] body, System.Action`2[T1,T2] bodyWithState, System.Action`3[T1,T2,T3] bodyWithStateAndIndex, System$
  at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal] (System.Collections.Generic.IEnumerable`1[T] source, System.Threading.Tasks.ParallelOptions parallelOptions, System.Action`1[T] body, System.Action`2[T1,T2] bodyWithState, System.Action`3[T1,T$
  at System.Threading.Tasks.Parallel.ForEach[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Action`1[T] body) [0x0001c] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at Eco.World.WorldPlugin.Run () [0x00026] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at Eco.Server.PluginManager+<>c__DisplayClass18_0.<StartPlugins>b__0 (System.Object x) [0x00000] in <170e11321e414586b2de019f749a752b>:0
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00025] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.ThreadHelper.ThreadStart (System.Object obj) [0x0000f] in <997dde78663c4b4bbf4a1f2b8d296905>:0
---> (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object
  at Eco.World.PersistentChunk.Tick () [0x00057] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at Eco.World.PersistentChunk.TryTick () [0x00008] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at Eco.World.WorldPlugin+<>c__DisplayClass24_0.<Run>b__0 (Eco.World.PersistentChunk chunk) [0x0000c] in <27e2c859b4d8401d9a5604af31cd0aeb>:0
  at System.Threading.Tasks.Parallel+<>c__DisplayClass30_0`2[TSource,TLocal].<ForEachWorker>b__0 (System.Int32 i) [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Parallel+<>c__DisplayClass17_0`1[TLocal].<ForWorker>b__1 () [0x0020b] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.InnerInvoke () [0x0000f] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task.InnerInvokeWithArg (System.Threading.Tasks.Task childTask) [0x00000] in <997dde78663c4b4bbf4a1f2b8d296905>:0
  at System.Threading.Tasks.Task+<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0 (System.Object <p0>) [0x00086] in <997dde78663c4b4bbf4a1f2b8d296905>:0 <---


--END DUMP--

```

**Comment by metachronism** [comment]
fixed for 7.0.2
its the same bug linked in another report
